src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -159,15 +161,19 @@ def set_package_install_status(self, package_names, package_versions, status="Pe
         package_names, package_versions = self.validate_packages_being_installed(package_names, package_versions)
         package_install_status_summary = ""
 
+        # lazy initialize OrderedMap, problem can't initialize any objects in __init__
+        if self.__installation_packages_map is None:
+            self.__installation_packages_map = OrderedDict()
+
         for package_name, package_version in zip(package_names, package_versions):
             patch_already_saved = False
             patch_id = self.__get_patch_id(package_name, package_version)
-            for i in range(0, len(self.__installation_packages)):
-                if patch_id == self.__installation_packages[i]['patchId']:
-                    patch_already_saved = True
-                    if classification is not None:
-                        self.__installation_packages[i]['classifications'] = [classification]
-                    self.__installation_packages[i]['patchInstallationState'] = status
+            # Match patch_id in map and update existing patch's classification i.e from None -> security and update pending status
+            if len(self.__installation_packages_map) > 0 and patch_id in self.__installation_packages_map:
+                if classification is not None:
+                    self.__installation_packages_map.setdefault(patch_id, {})['classifications'] = [classification]
+                self.__installation_packages_map.setdefault(patch_id, {})['patchInstallationState'] = status
```

Comment:
L 174 is marked is marked to be missing code coverage

Reply:
move logic to init and test cover